### PR TITLE
Ballgag refactor and conversion to GAGS

### PR DESCRIPTION
**Pull request by tf-4**
## About The Pull Request

Title.

The sizing is cycled when used in-hand, rather than a radial menu. The alternate colours have been removed in favour of converting it all to GAGS. Proper GAGS support in the kink-vends will be atomised soon(tm).

image

## Changelog

:cl:
refactor: Ballgags have been refactored, including conversion to GAGS
del: The kazoo gag has been removed
/:cl:

### CODE REVIEW

File: modular_skyrat/modules/modular_items/lewd_items/code/lewd_clothing/ballgag.dm [hunk 1]
```diff
@@ -1,211 +1,123 @@
...
-	gag_customizability = GAG_COLORABLE_AND_SIZABLE
-
-// Basically just a kazoo that goes in the moan hole
-/obj/item/clothing/mask/ballgag/phallic/kazoo
-	name = "phallic kazoo gag"
-	desc = "While it doesn't prevent the wearer from breathing, it does make everything they say sound quite silly."
-	moans = list("BVVV...", "VRRRRH...", "VRRR...", "VZZZ...", "ZYYY...", "BZZZ...")
-	moans_alt = list("WHOOONK!", "VAAAAMP!", "BRRRVVV!")
-	moan_sounds = list('modular_skyrat/modules/modular_items/lewd_items/sounds/kazoo_1.ogg',
-						'modular_skyrat/modules/modular_items/lewd_items/sounds/kazoo_2.ogg',
```
Comment (ORCACommander) [inline modular_skyrat/modules/modular_items/lewd_items/code/lewd_clothing/ballgag.dm:62]: if nothing else references these sound files, they should be deleted too. unused sound files eat bandwidth like candy

Comment (tf-4) [reply modular_skyrat/modules/modular_items/lewd_items/code/lewd_clothing/ballgag.dm:62]: Ah you're right, entirely slipped my mind. Ty

**Comment by tf-4** [comment]
Verified ✔️

**Review by GoldenAlpharex — APPROVED** [review]
Looks good to me.

**Comment by Lamella-0587** [comment]
Uhm. Hi! I'm the original creator of the ball gags item. During developing this PR you might accidentally remove some of my features, like ballgag size and good color palette with hue shifting. I understand the idea of creating a custom ball gag since I have the same thing with custom and classic dildo items, but it's not nice to remove features. Can you, please, add it as a "custom" model? A hue-shifted palette is really important for good graphics look.

**Comment by tf-4** [comment]
> Uhm. Hi! I'm the original creator of the ball gags item. During developing this PR you might accidentally remove some of my features, like ballgag size and good color palette with hue shifting. I understand the idea of creating a custom ball gag since I have the same thing with custom and classic dildo items, but it's not nice to remove features. Can you, please, add it as a "custom" model? A hue-shifted palette is really important for good graphics look.

Nothing was removed

**Pull request merged**